joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -110,11 +123,25 @@ trait AstForExpressionsCreator { this: AstCreator =>
     scope.pushNewBlockScope(blockNode)
     localAstParentStack.push(blockNode)
 
+    val receiverNode = astForNodeWithFunctionReference(callee)
+    Ast.storeInDiffGraph(receiverNode, diffGraph)
+
+    val typeHint = codeOf(receiverNode.nodes.head)
+    registerType(typeHint, typeHint)
+
+    receiverNode.nodes.head match {
+      case i: NewIdentifier =>
+        i.typeFullName = typeHint
+        i.dynamicTypeHintFullName = ArraySeq(typeHint)
+      case _ =>
+    }
+
     val tmpAllocName      = generateUnusedVariableName(usedVariableNames, "_tmp")
-    val localTmpAllocNode = createLocalNode(tmpAllocName, Defines.ANY.label)
+    val localTmpAllocNode = createLocalNode(tmpAllocName, typeHint).dynamicTypeHintFullName(ArraySeq(typeHint))
```

Comment:
I think we can drop this part. With the `<operator>.new` in place there's no longer a reason do this in the frontend. It would be easy to do the exact same thing in a pass - and possible to do arbitrarily more complex things.